### PR TITLE
Allow standard field default value and settings editing (#7104)

### PR DESCRIPTION
**Pull request by ad-elias**

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/boolean/components/SettingsDataModelFieldBooleanForm.tsx [hunk 1]
```diff
@@ -44,9 +42,6 @@ export const SettingsDataModelFieldBooleanForm = ({
           value={value}
           onChange={onChange}
           selectClassName={className}
-          // TODO: temporary fix - disabling edition because after editing the defaultValue,
-          // newly created records are not taking into account the updated defaultValue properly.
```
Comment (ad-elias) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/boolean/components/SettingsDataModelFieldBooleanForm.tsx:48]: Could not replicate this issue anymore so removed the comment

**Comment by ijreilly** [comment]
@ad-elias can you ping me when this is ready for review again? thanks :)

**Comment by ad-elias** [comment]
@ijreilly Ready for review!

**Review by ijreilly — APPROVED** [review]
great work! thanks a lot! 
(added a little commit to include Number fields that had been forgotten i think).

**Pull request merged**